src/NServiceBus.Core/Licensing/LicenseManager.cs

```diff
@@ -2,37 +2,81 @@ namespace NServiceBus
 {
     using System;
     using System.Diagnostics;
-#if NETSTANDARD
-    using System.Runtime.InteropServices;
-#endif
     using System.Text;
     using System.Threading;
     using System.Threading.Tasks;
     using Logging;
     using Particular.Licensing;
+#if NETSTANDARD
+    using System.Runtime.InteropServices;
+
+#endif
 
     class LicenseManager
     {
         internal bool HasLicenseExpired => result?.HasExpired ?? true;
 
+        public LicenseManager(Func<DateTime> utcDateTimeProvider)
+        {
+            this.utcDateTimeProvider = utcDateTimeProvider;
+        }
+
         internal void InitializeLicense(string licenseText, string licenseFilePath)
         {
             var licenseSources = LicenseSources.GetLicenseSources(licenseText, licenseFilePath);
 
             result = ActiveLicense.Find("NServiceBus", licenseSources);
 
             LogFindResults(result);
-
             if (result.HasExpired)
             {
-                if (result.License.IsTrialLicense)
+                LogExpiredLicenseError(result.License, Logger);
+            }
+            else
+            {
+                LogExpiringLicenseWarning(result.License, Logger);
+            }
```

Comment:
When I first read these conditions, I got confused, because it looks like we are logging an error only when the license has expired, but we are logging a warning _always_. But when I looked inside `LogExpiringLicenseWarning()` I realised that it doesn't do what it says on the tin. It only logs under specific conditions. I guess that method could just be renamed, but it may be be better to restructure this.

My usual approach to is to first inline both methods, then step back and look at what, if anything, should be extracted into a method or methods.

Reply:
I had the method inlined first, but `result.HasExpired` is not unit-testable as it's code is using `DateTime.UtcNow`. This structure is mostly for unit testing purposes. Happy for suggestions on better naming though. `TryLogExpiringLicenseWarning` maybe?

Reply:
The `TryXXX` convention is usually for methods that attempt the stated action and return a `bool` to indicate success. If it's just going to be a rename, it should convey "log if expiring". The code smell is that it's often difficult to name a method in that way.